src/ViewModel/Converter/ArticleRelatedItemTeaserConverter.php

```diff
@@ -19,11 +21,15 @@ final class ArticleRelatedItemTeaserConverter implements ViewModelConverter
 
     private $viewModelConverter;
     private $urlGenerator;
+    private $authorizationChecker;
+    private $rdsArticles;
 
-    public function __construct(ViewModelConverter $viewModelConverter, UrlGeneratorInterface $urlGenerator)
+    public function __construct(ViewModelConverter $viewModelConverter, UrlGeneratorInterface $urlGenerator, AuthorizationCheckerInterface $authorizationChecker, array $rdsArticles)
     {
         $this->viewModelConverter = $viewModelConverter;
         $this->urlGenerator = $urlGenerator;
+        $this->authorizationChecker = $authorizationChecker;
```

Comment:
a bit concerned that this feature flag is leaking everywhere, whereas we would like to make it easy to remove later on. Let me think about it

Reply:
Wonder if something like https://github.com/elifesciences/journal/blob/develop/src/Guzzle/SubjectRewritingMiddleware.php is needed instead...

Reply:
> Wonder if something like https://github.com/elifesciences/journal/blob/develop/src/Guzzle/SubjectRewritingMiddleware.php is needed instead...

I've update the PR with this approach. If we are happy then I will introduce a test for the middleware.